File: bro-common/src/main/java/me/xx2bab/bro/common/util/FileUtils.java
package me.xx2bab.bro.common.util;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileInputStream;
import java.io.FileWriter;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;

public class FileUtils {

    private static final FileUtils defaultInstance = new FileUtils();

    public static FileUtils getDefault() {
        return defaultInstance;
    }

    public String filterIllegalCharsForResFileName(String origin) {
        return origin.replace(":", "_")
                .replace(".", "_")
                .replace("-", "_");
    }

    public String readFile(File file) {
        try {
            InputStream inputStream = new FileInputStream(file);
            BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
            StringBuilder builder = new StringBuilder();
            String line;
            while ((line = reader.readLine()) != null) {
                builder.append(line);
            }
            return builder.toString();
        } catch (Exception e) {
            System.out.println(e.getMessage());
            return null;
        }
    }

    public void writeFile(String content, String filePath, String fileName) {
        File folder = new File(filePath);
        if (!folder.exists()) {
            boolean result = folder.mkdirs();
            if (!result) {
                return;
            }
        }
        try {
            File file = new File(filePath + File.separator + fileName);
            if (!file.exists()) {
                boolean result = file.createNewFile();
                if (!result) {
                    return;
                }
            }

            FileWriter writer = new FileWriter(file);
            writer.write(content);
            writer.close();
        } catch (IOException e) {
            System.out.println(e.getMessage());
        }
    }

}

File: bro-common/src/main/kotlin/me/xx2bab/bro/common/util/FileUtils.kt
package me.xx2bab.bro.common.util

import java.io.BufferedReader
import java.io.File
import java.io.FileWriter
import java.io.IOException

class FileUtils {
    fun filterIllegalCharsForResFileName(origin: String): String {
        return origin.replace(":", "_")
                .replace(".", "_")
                .replace("-", "_")
    }

    fun readFile(file: File?): String? {
        try {
            if (file == null) {
                return null
            }
            val bufferedReader: BufferedReader = file.bufferedReader()
            return bufferedReader.use { it.readText() }
        } catch (e: Exception) {
            println(e.message)
            return null
        }
    }

    fun writeFile(content: String?, filePath: String?, fileName: String?) {
        if (filePath == null || fileName == null) {
            return
        }
        val folder = File(filePath)
        if (!folder.exists()) {
            val result = folder.mkdirs()
            if (!result) {
                return
            }
        }
        try {
            val file = File(filePath + File.separator + fileName)
            if (!file.exists()) {
                val result = file.createNewFile()
                if (!result) {
                    return
                }
            }
            val writer = FileWriter(file)
            writer.write(content ?: "")
            writer.close()
        } catch (e: IOException) {
            println(e.message)
        }
    }

    companion object {
        val default = FileUtils()
    }
}